packages/vite/src/node/plugins/asset.ts

```diff
@@ -223,21 +225,26 @@ async function fileToBuiltUrl(
     // https://bundlers.tooling.report/hashing/asset-cascade/
     // https://github.com/rollup/rollup/issues/3415
     const map = assetHashToFilenameMap.get(config)!
-
     const contentHash = getAssetHash(content)
+    const { search, hash } = parseUrl(id)
+    const postfix = (search || '') + (hash || '')
+    const basename = path.basename(file)
+    const ext = path.extname(basename)
+    const fileName = path.posix.join(
+      config.build.assetsDir,
+      `${basename.slice(0, -ext.length)}.${contentHash}${ext}`
+    )
     if (!map.has(contentHash)) {
-      const basename = path.basename(file)
-      const ext = path.extname(basename)
-      const fileName = path.posix.join(
-        config.build.assetsDir,
-        `${basename.slice(0, -ext.length)}.${contentHash}${ext}`
-      )
       map.set(contentHash, fileName)
+    }
+    const emittedSet = emittedHashMap.get(config)!
```

Comment:
This is unsafe in terms of if someone changes the initialization in the future but didn't know that this non-null assertion extists
On the one hand this may be covered by tests, on the other hand we could just check the `emittedSet` explicitly here and prevent a `Uncaught ReferenceError: emittedSet is not defined`

Reply:
I think using `!` is ok here, as it implies that we are aware that `emittedHashMap.get(config)!` should always be present at this point. That ReferenceError looks good to me if someone wrongly changes the init code.

Reply:
But keep in mind that this would be a runtime error and not a compile time error